src/lambda_streams/Animals.java

```diff
@@ -32,31 +34,54 @@ public static void main(String[] args) {
 
     static List<String> capsFirst(List<String> animaList, boolean mutate) {
         //clean up the animals list so that the first letter is capitalized, and all the other letters are lowercased. Use a stream to accomplish this task.  Also, the 2nd parameter of this function is a boolean.  use this boolean 'flag' to determine whether or not to 'mutate' the original animals array stored as a static class field.  if the flag is set to 'true', mutate the animals and return the animals out of the function.  if it is false, create a copy of the animals, perform your stream operations on the copy, and return the copy of animals out of the function, WITHOUT modifying the original animals array.  
-        return animals;
+        if(mutate) {
+            return animals.stream().map(name -> Character.toUpperCase(name.charAt(0)) + name.substring(1).toLowerCase()).collect(Collectors.toList());
+        }
+        List<String> newAnimals = new ArrayList(animals);
+        return newAnimals.stream().map(name -> Character.toUpperCase(name.charAt(0)) + name.substring(1).toLowerCase()).collect(Collectors.toList());
     }
 
     static String addAnimal(String animal) {
-        //add an animal to the animal list.
+        animals.add(animal);
```

Comment:
Unless 'animals' is replaced with an ArrayList, this line of code fails with 'Exception in thread "main" java.lang.UnsupportedOperationException'.